### PR TITLE
fix: replaced deprecated UNSAFE_componentWillReceiveProps with componentDidUpdate

### LINKED ISSUE #1794: Warning: Using UNSAFE_componentWillReceiveProps in strict mode is not recommended
Hi!

I have an old bug [see here](https://github.com/rjsf-team/react-jsonschema-form/issues/1436) but i'm with the last version.

I know that the warning is not raised in production but normally this bug was fixed in the 2.0 version, isn't it ?

I'm only reusing one Form component and only change data and uischema for each schema I have in my multistep form.

### Description

index.js:1 Warning: Using UNSAFE_componentWillReceiveProps in strict mode is not recommended and may indicate bugs in your code. See https://fb.me/react-unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state

Please update the following components: Form

### Version
react: 16.13.1
@rjsf/core: 2.0.1

Cheers,
Yu.

### PR DESCRIPTION
**Pull request by lrozewicz**
### Reasons for making this change

I have replaced the deprecated UNSAFE_componentWillReceiveProps() method in the Form.tsx component with an improved solution utilizing the React lifecycle methods: getSnapshotBeforeUpdate() and componentDidUpdate().

fixes #1794 

### Checklist

- [ ] **I'm updating documentation**
- [ ] I've [checked the rendering](https://rjsf-team.github.io/react-jsonschema-form/docs/contributing) of the Markdown text I've added
- [x] **I'm adding or updating code**
- [x] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
- [ ] I've updated [docs](https://rjsf-team.github.io/react-jsonschema-form/docs) if needed
- [x] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/main/CHANGELOG.md) with a description of the PR
- [ ] **I'm adding a new feature**
- [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/core/src/components/Form.tsx [hunk 1]
```diff
@@ -280,22 +280,60 @@ export default class Form<
...
+   * it applies the next state and, if needed, triggers the `onChange` handler to inform about changes.
+   *
+   * This method effectively replaces the deprecated `UNSAFE_componentWillReceiveProps`, providing a safer alternative to handle prop changes and state updates.
+   *
+   * @param _ - The previous set of props.
+   * @param prevState - The previous state of the component before the update.
+   * @param snapshot - The value returned from `getSnapshotBeforeUpdate`.
+   */
+
+  componentDidUpdate(
```
Comment (heath-freenome) [inline packages/core/src/components/Form.tsx:320]: Please remove the blank line between the documentation and the function
```suggestion
   */
  componentDidUpdate(
```

File: packages/core/src/components/Form.tsx [hunk 2]
```diff
@@ -280,22 +280,60 @@ export default class Form<
...
-  UNSAFE_componentWillReceiveProps(nextProps: FormProps<T, S, F>) {
-    const nextState = this.getStateFromProps(nextProps, nextProps.formData);
-    if (
-      !deepEquals(nextState.formData, nextProps.formData) &&
-      !deepEquals(nextState.formData, this.state.formData) &&
-      nextProps.onChange
-    ) {
-      nextProps.onChange(nextState);
+
+  getSnapshotBeforeUpdate(
```
Comment (heath-freenome) [inline packages/core/src/components/Form.tsx:297]: Please remove the blank line between the documentation and the function
```suggestion
  getSnapshotBeforeUpdate(
```

File: packages/core/test/Form.test.jsx [hunk 3]
```diff
@@ -3016,17 +3016,20 @@ describeRepeated('Form common', (createFormComponent) => {
...
+            message: 'must match format "area-code"',
+            name: 'format',
+            params: { format: 'area-code' },
+            property: '.areaCode',
+            schemaPath: '#/properties/areaCode/format',
+            stack: '.areaCode must match format "area-code"',
+          },
+        ]);
+      }, 0);
+      // We use setTimeout with a delay of 0ms to allow all asynchronous operations to complete in the React component. Even though this is an "around" approach, it turned out to be the only effective method to handle this particular test case.
```
Comment (heath-freenome) [inline packages/core/test/Form.test.jsx:3032]: Two things:
1. I believe by `"around" approach` you mean `"round-about" approach` or `"indirect" approach`?
2. Also, can you keep the length of the comment to 120 character (per our formatting rules)? Feel free to make the comment be multiple lines

Update all these identical comments please when you make the change

**Comment by heath-freenome** [comment]
Looks good otherwise! Thanks for doing this!

File: packages/core/src/components/Form.tsx (hunk 1 above)
Comment (lrozewicz) [reply packages/core/src/components/Form.tsx]: done

File: packages/core/src/components/Form.tsx (hunk 2 above)
Comment (lrozewicz) [reply packages/core/src/components/Form.tsx]: done

File: packages/core/test/Form.test.jsx (hunk 3 above)
Comment (lrozewicz) [reply packages/core/test/Form.test.jsx]: ad. 1) I mean "round-about" approach
ad. 2) done

**Review by heath-freenome — APPROVED** [review]

**Pull request merged**